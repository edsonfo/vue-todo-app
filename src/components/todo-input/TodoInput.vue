<template>
  <form class="input-model">
    <v-text-field
      class="text-placeholder"
      placeholder="what needs to be done?"
      clearable
      v-model="textValue"
    >
      <v-btn icon slot="prepend-inner">
        <v-icon>
          {{ icons.mdiChevronDown }}
        </v-icon>
      </v-btn>

      <v-icon @click="onAddTodo" v-if="!!textValue" icon slot="append">
        {{ icons.mdiContentSave }}
      </v-icon>
    </v-text-field>
  </form>
</template>

<script>
  import { mdiChevronDown, mdiContentSave } from '@mdi/js'

  export default {
    name: 'TodoInput',

    data() {
      return {
        textValue: '',
        icons: {
          mdiChevronDown,
          mdiContentSave
        }
      }
    },
    methods: {
      onAddTodo() {
        this.$emit('add-todo')
      }
    }
  }
</script>
<style lang="less" scoped>
  .input-model {
    max-width: 50%;
    margin-right: auto;
    margin-left: auto;
  }
  .text-placeholder {
    padding: 16px 16px 16px 16px;
    font-family: Roboto;
    font-style: oblique;
    font-size: 26px;
    max-width: 1300px;
    margin-bottom: -18px;
    min-width: 600px;
  }
</style>
