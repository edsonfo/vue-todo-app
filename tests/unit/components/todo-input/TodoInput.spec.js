import { shallowMount } from '@vue/test-utils'
import TodoInput from '@/components/todo-input/TodoInput.vue'

describe('TodoInput component tests', () => {
  const wrapper = shallowMount(TodoInput)
  it('should match snapshot', () => {
    expect(wrapper.element).toMatchSnapshot()
  })
})
