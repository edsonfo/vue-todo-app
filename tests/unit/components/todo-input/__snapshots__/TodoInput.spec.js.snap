// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoInput component tests should match snapshot 1`] = `
<form
  class="input-model"
>
  <v-text-field-stub
    backgroundcolor=""
    class="text-placeholder"
    clearable="true"
    clearicon="$clear"
    errorcount="1"
    errormessages=""
    loaderheight="2"
    messages=""
    placeholder="what needs to be done?"
    rules=""
    successmessages=""
    type="text"
    value=""
  >
    <v-btn-stub
      activeclass=""
      icon="true"
      tag="button"
      type="button"
    >
      <v-icon-stub>
        
        M7.41,8.58L12,13.17L16.59,8.58L18,10L12,16L6,10L7.41,8.58Z
      
      </v-icon-stub>
    </v-btn-stub>
     
    <!---->
  </v-text-field-stub>
</form>
`;
